### ISSUE TITLE
3.0.6 - js uglification?

### ISSUE THREAD

**Issue by zyro23** [bertramdev/asset-pipeline#232]
seems like 3.0.6 does minify js in terms of whitespace but it does not uglify it.

if i understand it right,
* babel transpiles, but does not minify/uglify (`babel.options = '{"presets": ["minify"]}'` says preset not found)
* closure-compiler does only uglify/rename variables if it actually transpiles...

unfortunately, the current evaluation of the closure compiler language modes (in and out) does only allow `LanguageMode.ECMASCRIPT_NEXT`. if `languageIn` and `languageOut` are the same, we get whitespace minification without any variable renaming, it seems.

would be great to have some kind of uglification functionality back.
while i think i would prefer babel support for the `minify` preset, i do not know if that works at all on the jvm without a node environment, and if it does, what the performance looks like.

thank you

**Comment by aibolit** [comment]
I definitely would like some sort of minification as well. 
It looks like the there is a [babel standalone](https://babeljs.io/docs/en/next/babel-standalone.html) file located [here](https://github.com/bertramdev/asset-pipeline/blob/master/asset-pipeline-core/src/main/resources/asset/pipeline/babel.min.js) and invoked the Nashorn engine built into the jvm [here](https://github.com/bertramdev/asset-pipeline/blob/master/asset-pipeline-core/src/main/groovy/asset/pipeline/processors/BabelJsProcessor.groovy#L53-L62).

If a different version of the babel-standalone could be included that supported all of the presets it would just work. The question is where to find this.

**Comment by davydotcom** [comment]
Restored closures default minify configuration to assume a target and source language on the post processor of ECMASCRIPT 5 since babel transpiles to that anyway. If one wishes to have asset-pipeline use ECMASCRIPT_NEXT it can be set in the minifyOptions in `build.gradle` Simply set 

```
assets {
  minifyOptions = [languageMode: 'ECMASCRIPT_NEXT']
}
```

**Comment by davydotcom** [comment]
This will be resolved in 3.2.3
